wallet-new/src/Cardano/Wallet/Kernel/Transactions.hs

```diff
@@ -213,22 +253,22 @@ instance Arbitrary EstimateFeesError where
     arbitrary = EstFeesTxCreationFailed <$> arbitrary
 
 estimateFees :: ActiveWallet
-             -> IO Address
-             -- ^ A computation to generate a new change 'Address.
-             -> (Address -> Either CoinSelHardErr SafeSigner)
-             -- ^ A function to sign each Address
+             -> PassPhrase
+             -- ^ The spending password.
              -> CoinSelectionOptions
              -- ^ The options describing how to tune the coin selection.
              -> HdAccountId
-             -- ^ The source @(root, account) from where the payment was
-             -- originated
+             -- ^ The source HD Account from where the payment should originate
              -> NonEmpty (Address, Coin)
              -- ^ The payees
              -> IO (Either EstimateFeesError Coin)
-estimateFees activeWallet@ActiveWallet{..} genChangeAddr signAddress options accountId payees = do
+estimateFees activeWallet@ActiveWallet{..} spendingPassword options accountId payees = do
     snapshot         <- getWalletSnapshot walletPassive
```

Comment:
TODO: have `newTransaction` returns the utxo it originally used so that the calculation of the fee is more precise.